components/nodemanager-service/pgdb/project_update.go

```diff
@@ -0,0 +1,281 @@
+package pgdb
+
+import (
+	"context"
+
+	"encoding/json"
+
+	iam_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	"github.com/chef/automate/components/nodemanager-service/api/nodes"
+	project_update_lib "github.com/chef/automate/lib/authz"
+	"github.com/chef/automate/lib/stringutils"
+	uuid "github.com/chef/automate/lib/uuid4"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const selectNodesProjectData = `
+SELECT
+  n.id,
+  n.projects_data
+FROM nodes n
+`
+
+type NodeProjectData struct {
+	ID                   string `db:"id"`
+	Data                 string `db:"projects_data"`
+	ProjectDataKeyValues []*nodes.ProjectsData
+}
```

Comment:
The "JobCancel", "UpdateProjectTags", and "JobStatus" are the three functions that are called from the cereal project update service.